pyrsistent/_pclass.py

```diff
@@ -135,7 +135,14 @@ def transform(self, *transformations):
         return transform(self, transformations)
 
     def __eq__(self, other):
+        if self is other:
+            return True
         if isinstance(other, self.__class__):
+            try:
```

Comment:
Can we break this down further?

    def __eq__(self, other):
          if self is other:
               return True
         if not isinstance(other, self.__class__):
               return False
         try:
              self_hash, other_hash = hash(self), hash(other)
         except:
              pass
         else:
              if self_hash != other_hash:
                   return False
          for name in .....

This way you only protect the minimum amount of lines with try/except: